test/Base_Tests/src/Semantic/Arrow_Interop_Spec.enso

```diff
@@ -0,0 +1,40 @@
+from Standard.Base import all
+import Standard.Base.Errors.Common.Index_Out_Of_Bounds
+import Standard.Base.Errors.Common.Unsupported_Argument_Types
+import Standard.Base.Errors.Common.Not_Invokable
+import Standard.Base.Errors.Common.No_Such_Method
+
+from Standard.Test import all
+
+foreign arrow new_arrow = """
+    new[Int8]
+
+
+add_specs suite_builder =
+    pending = if Polyglot.is_language_installed "arrow" then Nothing else """
+        Can't run Arrow tests, Arrow is not installed.
+
+    suite_builder.group "Arrow" pending=pending group_builder->
+        group_builder.specify "should allow for creating a new fixed Array in Arrow" <|
+            arrow_array_constructor = new_arrow
+            builder = arrow_array_constructor.new 10
+
+            builder.length . should_equal 10
+            builder.at 1 . should_equal Nothing
+            builder.at 10 . should_fail_with Index_Out_Of_Bounds
+            builder.append 2
+            builder.append 10
+            builder.at 1 . should_equal 10
+            builder.append 127
+
+            # Check that the value that will not fit into a byte is rejected
+            Test.expect_panic_with (builder.append 128) Not_Invokable
```

Comment:
If you throw [UnsupportedTypeException](https://www.graalvm.org/truffle/javadoc/com/oracle/truffle/api/interop/UnsupportedTypeException.html) in the `builder.writeArrayElement` interop message, then (I hope) you'll get better Enso exception than `Not_Invokable` here.

Reply:
Nope. Invoke can't throw `UnsupportedTypeException` so I have to wrap it in `UnsupportedMessageException` anyway.

Reply:
At https://github.com/enso-org/enso/pull/9150/files#diff-ebe312b126e000670288fc461a3e77fa0551b8e32341d54643d0a5e494f7446bR31 I see:
```
 Test.expect_panic_with (builder.append 128) Unsupported_Argument_Types
```
that's a reasonable exception type, thank you. However at https://github.com/enso-org/enso/pull/9150/files#diff-ebe312b126e000670288fc461a3e77fa0551b8e32341d54643d0a5e494f7446bR34 I can still find:
```
Test.expect_panic_with (builder.append 21) No_Such_Method
``` 
I find that misleading.